lib/matplotlib/pyplot.py

```diff
@@ -2180,9 +2180,13 @@ def set_cmap(cmap):
     matplotlib.cm.register_cmap
     matplotlib.cm.get_cmap
     """
+    # The name of the colormap may be different than the registered name.
+    # We want the registered name for the rc setting
+    name = getattr(cmap, "name", cmap)
+    # Run this through get_cmap to confirm this is a valid cmap before
```

Comment:
Can this be clearer which "this" is meant here?  I almost commented based on this comment that the next line should be `cmap = get_cmap(name)`, but am no longer sure which was intended.

Reply:
I _think_ we want to use `get_cmap(cmap)` here in case someone didn't register the colormap? Although that would also mean the rc paramter wouldn't work then either...